### PR TITLE
[test] Annotate an expected test crash for humans who read crash logs.

### PR DESCRIPTION
**Pull request by gparker42**

### CODE REVIEW

**Comment by gparker42** [comment]
@swift-ci please smoke test

**Comment by shahmishal** [comment]
@swift-ci please smoke test

File: lib/SILOptimizer/UtilityPasses/BugReducerTester.cpp [hunk 1]
```diff
@@ -58,6 +58,12 @@ static llvm::cl::opt<FailureKind> TargetFailureKind(
         clEnumValEnd),
     llvm::cl::init(FailureKind::None));
 
+
+LLVM_ATTRIBUTE_NOINLINE
+void THIS_TEST_IS_EXPECTED_TO_CRASH_HERE() {
+  llvm_unreachable("Found the target!");
```
Comment (jckarter) [inline lib/SILOptimizer/UtilityPasses/BugReducerTester.cpp]: `assert(false && "...");` or `abort();` would be more appropriate if you want to ensure there's a real crash here and not just a fall-off-the-edge.

Comment (gparker42) [inline lib/SILOptimizer/UtilityPasses/BugReducerTester.cpp]: @gottesmm, do you want to tweak that?

**Pull request merged**